### PR TITLE
Dyn 3513 preferences modal dialog

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Create a new modal dialog for the new Preferences window.
The requirements are the next ones:

- When user press preferences... in Dynamo menu under debug mode, the dialog appears
- A new modal dialog is created based on the design by UX
   - The modal dialog should come with Preferences Title
   - The modal dialog should come with Save Changes button
   - The modal dialog should come with split lines
- Dynamo will be blocked until user exit that dialog

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang 

### FYIs

@Astul-Betizagasti @alfredo-pozo

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
Here is a gif of the new Preferences window.
dynamo_preferences_menu

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+            <SolidColorBrush x:Key="PreferencesWindowButtonMouseOverFontColor" 
+                             Color="#CDCBCB" />
+            <SolidColorBrush x:Key="CloseButtonMouseOverBackground" 
+                             Color="#E81123" />
+            <SolidColorBrush x:Key="CloseButtonPressed" 
+                             Color="#B70F1B" />
+            <SolidColorBrush x:Key="CloseButtonMouseOverForeground" 
+                             Color="#FFFFFF" />
+            <SolidColorBrush x:Key="TabItemUnSelected" 
+                             Color="DarkGray" />
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:35]: Thanks! Check for Dynamo/src/DynamoCoreWpf/UI/Themes/Modern, I think most Dynamo styles are stored there, if possible I would prefer styles not stored in the modal dialog xaml. Would you look at it?

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+        xmlns:p="clr-namespace:Dynamo.Wpf.Properties;assembly=DynamoCoreWpf"
+        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+        xmlns:views="clr-namespace:Dynamo.Wpf.Views"
+        WindowStartupLocation="CenterOwner"
+        WindowStyle="None"
+        ResizeMode="NoResize"
+        mc:Ignorable="d" 
+        Height="474" 
+        Width="530" >
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:14]: Can you add some comments before each UI components in this xaml for future reference? Thanks!

Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:17]: @QilongTang I added extra comment in the PreferencesView.xaml. Thanks

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml]: @QilongTang I moved all the Styles to the src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml file. Thanks

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 3]
```diff
@@ -1320,6 +1327,7 @@
     <Resource Include="UI\Images\icon-whats-new-small.png" />
     <Resource Include="UI\Images\icon-dictionary-small.png" />
   </ItemGroup>
+  <ItemGroup />
```
Comment (QilongTang) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:1330]: This line seems not needed?

Comment (RobertGlobant20) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj]: @QilongTang This line was removed in the next commit: https://github.com/DynamoDS/Dynamo/pull/11534/commits/eb0d517fb2dd151c02a9c68afc082913668a6371

**Review by QilongTang — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@QilongTang question about the funky commits here, were these not squashed by accident or are all these commits showing up anyway for some reason?

**Comment by QilongTang** [comment]
> @QilongTang question about the funky commits here, were these not squashed by accident or are all these commits showing up anyway for some reason?

This is usually because of how developer leverage local fork, we can check with @RobertGlobant20 . I bet instead of pulling upstream, some client do merge commits so the fork head looks really messy with bunch of commits with same title `Merge branch 'master' of https://github.com/DynamoDS/Dynamo`

**Pull request merged**